### PR TITLE
✨ Fetch nft class listing on chain in buy NFT functions

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

**Review by nwingt — APPROVED** [review]

File: src/store/modules/nft.js [hunk 1]
```diff
@@ -272,11 +278,38 @@ const actions = {
...
+        } = l;
+        return {
+          classId,
+          nftId,
+          seller,
+          price: new BigNumber(price).shiftedBy(-9).toNumber(),
+          expiration: new Date(expiration),
+        };
+      })
+      .sort((a, b) => a - b)[0];
```
Comment (WeiJun0827) [inline src/store/modules/nft.js:302]:
```suggestion
      .sort((a, b) => a.price - b.price)[0];
```

File: src/store/modules/nft.js [hunk 2]
```diff
@@ -272,11 +278,38 @@ const actions = {
...
+  },
   async lazyGetNFTPurchaseInfo({ getters, dispatch }, classId) {
     let info = getters.getNFTClassPurchaseInfoById(classId);
+    const listingInfo = getters.getNFTClassListingInfoById(classId);
     if (!info) {
       info = await dispatch('fetchNFTPurchaseInfo', classId);
     }
+    if (!listingInfo) dispatch('fetchNFTListingInfo', classId);
     return info;
   },
```
Comment (WeiJun0827) [inline src/store/modules/nft.js:314]: Seems the `PurchaseInfo` term implies both the original `PurchaseInfo` and the new `ListingInfo`. Would a new term be considered to differentiate the purchase price from the new price that considers the listing price?

File: src/store/modules/nft.js [hunk 3]
```diff
@@ -441,6 +474,7 @@ const actions = {
             catchAxiosError(dispatch('fetchNFTPurchaseInfo', classId))
           );
         }
+        promises.push(dispatch('fetchNFTPurchaseInfo', classId));
```
Comment (WeiJun0827) [inline src/store/modules/nft.js:477]: Can this part be simplified?

File: src/store/modules/nft.js (hunk 2 above)
Comment (williamchong) [reply src/store/modules/nft.js:314]: `lazyGetNFTPurchaseAndListingInfo`?

File: src/store/modules/nft.js (hunk 3 above)
Comment (williamchong) [reply src/store/modules/nft.js:477]: You mean put this to line 465?

Comment (WeiJun0827) [reply src/store/modules/nft.js:477]: Yes, will the logic make sense?

File: src/store/modules/nft.js (hunk 2 above)
Comment (WeiJun0827) [reply src/store/modules/nft.js:314]: Yeah, looks good to me

File: src/store/modules/nft.js (hunk 3 above)
Comment (williamchong) [reply src/store/modules/nft.js]: Actually this should be `fetchNFTListingInfo`

**Review by WeiJun0827 — APPROVED** [review]

**Pull request merged**